PowerShellTools.HostService/ServiceManagement/Debugging/PowerShellDebuggingServiceEventHandlers.cs

```diff
@@ -116,14 +126,31 @@ private void Debugger_DebuggerStop(object sender, DebuggerStopEventArgs e)
                         file = _mapRemoteToLocal[bp.Script];
                     }
 
-                    _callback.DebuggerStopped(new DebuggerStoppedEventArgs(file, bp.Line, bp.Column));
+                    _callback.DebuggerStopped(new DebuggerStoppedEventArgs(true, file, bp.Line, bp.Column, false));
                 }
             }
             else
-            {
+            {                
                 if (_callback != null)
                 {
-                    _callback.DebuggerStopped(new DebuggerStoppedEventArgs());
+                    if (_attaching)

```

Comment:
You leave the "attaching process" as last case(last else) in GetCallStack, that implies there is a condition for that? I feel much better by using that condition than current flag implementation. The current flag implementation sets flags right before attaching, if the attaching work(enter-pshostprocess, or debug-runspace) throws, we end up with two messed status flags, which are going to constantly cause trouble to all debugging scenarios, and looks like the only way to clean them up is through detach(), however, since user did not successfully attach yet ...